src/js/index.js

```diff
@@ -59,28 +116,31 @@ function handleTags() {
 			el.classList.remove('visible');
 		}
 	});
-
-	if (!search) {
-		return;
-	}
-	search.set('tags', tags.join(' '));
-	const newRelativePathQuery = `${window.location.pathname}?${search.toString()}`;
-	history.pushState(null, '', newRelativePathQuery);
 }
 
 function handleSort() {
+	if (!sortParams.sort) {
+		sortParams.sort = sortingInput.value.split('_')[0];
+	}
+	if (!sortParams.order) {
+		sortParams.order = sortingInput.value.split('_')[1];
+	}
+	sortingInput.value = Object.values(sortParams).join('_');
 	Array.from(contentEl.children)
 		.map(child => contentEl.removeChild(child))
 		.sort(sort[sortingInput.value])
 		.forEach(sortedChild => contentEl.appendChild(sortedChild));
 }
 
 // The cascade is the function which handles calling filtering and sorting swag
-function cascade(force = false) {
+function cascade(force = false, firstRun = false) {
```

Comment:
Force should just be set to true on firstRun, that's basically what it's designed for